src/cursor/cur_table.c

```diff
@@ -433,22 +440,22 @@ __curtable_prev(WT_CURSOR *cursor)
 static int
 __curtable_reset(WT_CURSOR *cursor)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i;
 
     ctable = (WT_CURSOR_TABLE *)cursor;
-    /* Grab the primary cursor to reset the bounds. */
-    primary = *ctable->cg_cursors;
 
     JOINABLE_CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, reset, NULL);
 
     APPLY_CG(ctable, reset);
 
-    /* If a user calls cursor reset also reset the bounds. */
+    /* If a user calls cursor reset, call cursor reset on all column groups. */
     if (API_USER_ENTRY(session))
-        __wt_cursor_bound_reset(primary);
+        for (i = 0, cp = ctable->cg_cursors; i < WT_COLGROUPS(ctable->table); i++, cp++)
+            __wt_cursor_bound_reset(*cp);
```

Comment:
Is there any way we can make these three colgroup operations, `__wt_cursor_bound_reset` and `__wt_cursor_bounds_restore` and `cursor->bound` go through a macro?

Reply:
Leave this one with me for a while to think about.